Update OVERMAP.md (#46981)
CleverRaven/Cataclysm-DDA: doc/OVERMAP.md

made up of multiple overmap terrains (though not always), may have overmap connections (e.g. roads,
sewers, subways), and have JSON-defined rules guiding their placement.

### Special placement

There are a finite number of sectors in which overmap specials can be placed during overmap

> generation, each being a square with side length equal to `OMSPEC_FREQ` ( defined in omdata.h, at

Reviewer: ```suggestion
generation, each being a square with side length equal to `OMSPEC_FREQ` (defined in omdata.h; at
```

the time of writing `OMSPEC_FREQ`= 15 meaning each overmap has 144 sectors for placing specials). 
At the beginning of overmap generation, a list of eligable map specials is built 

> ( `overmap_special_batch` ). Next, a free sector is chosen where a special will be placed. A random

Reviewer: ```suggestion
(`overmap_special_batch`).  Next, a free sector is chosen where a special will be placed.  A random
```

> point in that sector is checked against a random rotation of a special from the special batch to see

Reviewer: Randomly chosen special?

Author: Yeah, random point in a random sector checked against a random rotation of a random special...very well randomized, all things considered. I'll update it to clear up potential confusion.

a specific rotated version of the associated overmap terrain--generally, this is the `_north` rotation
as it corresponds to the way in which the JSON for mapgen is defined.

### Connections

The overmap special can be connected to the road, subway or sewer networks. Specifying a connection
point causes the appropriate connection to be automatically generated from the nearest matching terrain
, unless `existing` is set to true. In that case the special can only be placed if the connection point
intersects an existing road/etc. Since the road network is sparse, and most roads will be generated to 

> connect up other specials this lowers the chances of the special spawning considerably.

Reviewer: ```suggestion
connect up other specials, this lowers the chances of the special spawning considerably.
```

### Occurrences (default)

Occurrences is the way to set the baseline rarity of a special. The field can behave in two ways:
by default, it sets the minimum and maximum occurrences of the special per overmap. Currently all
overmap specials have a minimum occurrence of 0, to keep the overmaps from being too similar to each
other. In addition, there are no specials with a maximum occurrence of 1. This is important because 
each normal special has a very high chance of being placed at least once per overmap, owing to some 

> quirks of the code ( the number of specials is only slightly higher than the number of slots per overmap,
> specials that failed placement don't get disqualified and can be rolled for again, placement iterating until
> all sectors are occupied ). For specials that are not common enough to warrant appearing on all overmaps

Reviewer: ```suggestion
quirks of the code (most notably, the number of specials is only slightly more than the number of slots per
overmap, specials that failed placement don't get disqualified and can be rolled for again, and placement iterates
until all sectors are occupied). For specials that are not common enough to warrant appearing on all overmaps
```

please use the "UNIQUE" flag.

### Occurrences ( UNIQUE )

When the special has the "UNIQUE" flag, instead of defining the minimum and maximum number placed 
the occurrences field defines the chance of the special to be included in any one given overmap.

> Before any placement rolls, all specials with this flag have to succeed in a x_in_y ( first value, second

Reviewer: ```suggestion
Before any placement rolls, all specials with this flag have to succeed in an x_in_y (first value, second
```

> value) roll to be included in the `overmap_special_batch` for the currently generated overmap, meaning that 

Reviewer: ```suggestion
value) roll to be included in the `overmap_special_batch` for the currently generated overmap;
```

> any special that failed this roll won't ever be considered for placement. Currently all UNIQUE specials
> use [ x, 100 ] occurrences for ease of understanding, but this is not binding.

Reviewer: ```suggestion
any special that failed this roll will never be considered for placement.  Currently all UNIQUE specials
use [x, 100] occurrences - percentages - for ease of understanding, but this is not required.
```

value for an individual entry takes precedence over the top level value, so you may define the top
level value and then only specify it for individual entries that differ.

### City distance and size 

During generation of a new overmap, cities and their connecting roads will be generated before 
specials are placed. Each city gets assigned a size at generation and will begin its life as a single 
intersection. The city distance field specifies the minimum and maximum distance the special can be 

> placed from this intersection, *not* from the edge of the city, meaning a special with a low minimum

Reviewer: ```suggestion
placed from _this_ intersection, *not* from the edge of the city, meaning a special with a low minimum
```

> distance and a relaxed city size restriction may be placed on the outer border of a larger city.

Reviewer: What is meant by a "relaxed city size restriction", and how does the city size field work, BTW? Only counts cities in that range for distance, or what?

Author: If the city size maximum is not smaller than the city distance minimum. All specials are considered to belong only to the nearest city, using the central intersection of that city as their origin.
